diff --git a/CMakeLists.txt b/CMakeLists.txt
index f16979a..5e71e57 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,7 +17,7 @@ find_package(yaml_cpp_vendor REQUIRED)
 find_package(yaml-cpp REQUIRED)
 
 find_package(gz_transport_vendor REQUIRED)
-find_package(gz-transport REQUIRED)
+find_package(gz-transport13 REQUIRED)
 
 find_package(gz_msgs_vendor REQUIRED)
 find_package(gz-msgs REQUIRED)
