diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a549c288d..5981e24ff 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -187,9 +187,9 @@ target_link_libraries(${gz_lib_target}
     gz-sim${PROJECT_VERSION_MAJOR}-gui
 )
 
-# Executable target that runs the GUI without ruby for debugging purposes.
-add_executable(runGui cmd/runGui_main.cc)
-target_link_libraries(runGui PRIVATE ${gz_lib_target})
+# # Executable target that runs the GUI without ruby for debugging purposes.
+# add_executable(runGui cmd/runGui_main.cc)
+# target_link_libraries(runGui PRIVATE ${gz_lib_target})
 
 # Create the library target
 gz_create_core_library(SOURCES ${sources} CXX_STANDARD 17)
